packages/bitcore-wallet-client/src/lib/payproV2.ts

```diff
@@ -319,7 +355,78 @@ export class PaymentProtocolV2Client {
       throw new Error('Response signature invalid');
     }
 
-    return { requestUrl, responseData, keyData };
+    return payProDetails;
+  }
+
+  /**
+   * Internal method for processing response
+   * @param {Object} responseData
+   * @return {Promise<Object{payProDetails: Object}>}
+   * @private
+   */
+
+  static processResponse(responseData) {
+
+    let payProDetails: any = {
+      payProUrl: responseData.paymentUrl,
+      memo: responseData.memo
+    };
+
+    // otherwise, it returns err.
+    payProDetails.verified = true;
+
+    // network
+    if (responseData.network == 'test')
+      payProDetails.network = 'testnet';
+
+    if (responseData.network == 'main')
+      payProDetails.network = 'livenet';
+
+    if (responseData.chain) {
+      payProDetails.coin = responseData.chain.toLowerCase();
+    }
+
+    if (responseData.expires) {
+      try {
+        payProDetails.expires = (new Date(responseData.expires)).toISOString();
+      } catch (e) {
+        throw new Error('Bad expiration');
+      }
+    }
+
+    // BTC Response - ETH Response
+    if (_.has(responseData, 'instructions[0].requiredFeeRate')) {
+      const requiredFeeRate = responseData.instructions[0].requiredFeeRate;
+      if (requiredFeeRate > MAX_FEE_PER_KB) {
+        throw new Error('Fee rate too high:' + requiredFeeRate);
+      }
+      payProDetails.requiredFeeRate = requiredFeeRate;
+    }
+    if (_.has(responseData, 'instructions[0].outputs[0].amount')) {
+      payProDetails.amount = responseData.instructions[0].outputs[0].amount;
+    }
+    if (_.has(responseData, 'instructions[0].outputs[0].address')) {
+      try {
+        const bitcore = Bitcore_[payProDetails.coin];
+        payProDetails.toAddress = (bitcore.Address(responseData.instructions[0].outputs[0].address)).toString(true);
+      } catch (e) {
+        return new Error('Bad output address ' + e);
+      }
+    }
+
+    // ETH Response
+    if (_.has(responseData, 'instructions[0].value')) {
```

Comment:
You're going to want to loop through the instructions and build a transaction per instruction, rather than get the first index.